src/espIdf/debugAdapter/debugAdapterManager.ts

```diff
@@ -81,34 +90,34 @@ export class DebugAdapterManager extends EventEmitter {
           "Invalid OpenOCD script path or access is denied for the user"
         );
       }
-      this.projectName = await getProjectName(this.currentWorkspace.fsPath);
-      const elfPath =
-        path.join(this.currentWorkspace.fsPath, "build", this.projectName) +
-        ".elf";
       const logFile = path.join(this.currentWorkspace.fsPath, "debug") + ".log";
 
       const pythonBinPath = idfConf.readParameter(
         "idf.pythonBinPath"
       ) as string;
-      this.adapter = spawn(
-        pythonBinPath,
-        [
-          this.debugAdapterPath,
-          "-d",
-          this.logLevel.toString(),
-          "-e",
-          elfPath,
-          "-l",
-          logFile,
-          "-om",
-          "connect_to_instance",
-          "-p",
-          this.port.toString(),
-          "-dn",
-          this.target,
-        ],
-        { env: this.env }
-      );
+      const adapterArgs = [
+        this.debugAdapterPath,
+        "-d",
+        this.logLevel.toString(),
+        "-e",
+        this.elfFile,
+        "-l",
+        logFile,
+        "-p",
+        this.port.toString(),
+        "-dn",
+        this.target,
+      ];
+      this.isPostMortemDebugMode
+        ? adapterArgs.push("-om", "without_oocd")
+        : adapterArgs.push("-om", "connect_to_instance");
+      if (this.isPostMortemDebugMode) {
+        adapterArgs.push("--postmortem"); // MODIFY for actual esp_dap_adapter argument
+      }
+      for (const setupCmd of this.setupCommands) {
+        adapterArgs.push("--setup-cmd", setupCmd.text); // MODIFY for actual esp_dap_adapter argument
```

Comment:
i'm going to use  `--gdbinit `cause we are usially use files with that name for these commands